holoviews/plotting/mpl/chart.py

```diff
@@ -1258,3 +1258,47 @@ class SideSpikesPlot(AdjoinedPlot, SpikesPlot):
         all axis labels including ticks and ylabel. Valid options are 'left',
         'right', 'bare' 'left-bare' and 'right-bare'.""")
 
+
+
+class SegmentPlot(ColorbarPlot):
+    """
+    Segments are lines in 2D space where each two key dimensions specify a
+    (x, y) node of the line.
+    """
+    style_opts = PathPlot.style_opts + ['cmap']
+
+    _nonvectorized_styles = ['cmap']
+
+    _plot_methods = dict(single='segment')
+
+    def init_artists(self, ax, plot_args, plot_kwargs):
+        if 'c' in plot_kwargs:
+            plot_kwargs['array'] = plot_kwargs.pop('c')
+        if 'vmin' in plot_kwargs and 'vmax' in plot_kwargs:
+            plot_kwargs['clim'] = plot_kwargs.pop('vmin'), plot_kwargs.pop('vmax')
+        line_segments = LineCollection(*plot_args, **plot_kwargs)
+        ax.add_collection(line_segments)
+        return {'artist': line_segments}
+
+    def get_data(self, element, ranges, style):
+        # Get [x0, y0, x1, y1, ...]
+        x0idx, y0idx, x1idx, y1idx = (
+            (1, 0, 3, 2) if self.invert_axes else (0, 1, 2, 3)
+        )
+        # Compute segments
+        dims = element.dimensions()
+        data = [[(x0, y0), (x1, y1)] for x0, y0, x1, y1
+                in element.array([x0idx, y0idx, x1idx, y1idx])]
```

Comment:
I think we need to unpack them all separate since ``.array`` may coerce types incorrectly.

Reply:
You're thinking of `element.dimension_values()`; or `element.array([x0idx])` etc?

Reply:
Either works, the former is more canonical (although I've wanted a short API for it for a long time).

Reply:
Done